modules/core/src/command_line_parser.cpp

```diff
@@ -9,6 +9,8 @@ static const char* noneValue = "<none>";
 
 static String cat_string(const String& str)
 {
+    if (str.length() == 0)
+        return String("");
     int left = 0, right = (int)str.length();
     while( left <= right && str[left] == ' ' )
```

Comment:
It is better to replace `<=` to `<` instead.

Also please remove unnecessary commits from PR.